src/features/unified-share-modal/utils/computeSuggestedCollabs.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @flow
+ * @file Suggested Collaborator utility functions
+ * @author Box
+ */
+import type { SuggestedCollab, SuggestedCollabLookup } from '../flowTypes';
+import fuzzySearch from '../../../utils/fuzzySearch';
+
+/**
+ * Function to compute suggested collaborators given a list of contacts and cached suggested collaborators.
+ * Used by input components to help augment API results with cached suggestions using a fuzzy search.
+ *
+ * @param {Array<C>} contacts Contacts returned from the server with exact matches
+ * @param {SuggestedCollabLookup} suggestedCollaborators Cached suggested collab lookup to fuzzy match on
+ * @param {string} inputValue Value the user typed into the input box searching for collabs
+ * @param {number} maxSuggestions Maximum number of items to suggest/return, default 3
+ * @param {number} minCharacters Minimum number of characters before searching, default 3
+ * @returns {Array<SuggestedCollab>, Array<C>}
+ * [0] Suggested collaborators in recommended order
+ * [1] The rest of the contacts not suggested
+ */
+export default function computeSuggestedCollabs<C: { id: number | string }>(
+    contacts: Array<C>,
+    suggestedCollabLookup: SuggestedCollabLookup,
+    inputValue: string,
+    maxSuggestions: number = 3,
+    minCharacters: number = 3,
+    maxGaps: number = 2,
+): [Array<SuggestedCollab>, Array<C>] {
+    const contactIdSet = new Set(contacts.map(c => c.id.toString()));
+    // $FlowFixMe
+    const suggestedCollabs: Array<SuggestedCollab> = Object.values(suggestedCollabLookup);
+    const exactMatches: Array<SuggestedCollab> = suggestedCollabs.filter(
+        sc => sc.id && contactIdSet.has(sc.id.toString()),
```

Comment:
a lot of id.toString() is being done. Can we change such that the incoming objects always has string ids from the absolute beginning in the component hierarchy? Ideally the backend or parent app's network call should be sending it that way.

Reply:
It seems like the api returns number for all the places I've seen. Is there a standard for what format we store the ids in on the frontend? I could change the existing contact id type to just number